### DISCUSSION TITLE
Adding windows authentication to Next Js Application.

### DISCUSSION THREAD

**Discussion by tpanw** [vercel/next.js#20597] [Help]
Hi Guys, I am new to Next Js and trying to add windows authentication to my application. I haven't found much material online for this. 

So far I have tried to use Node SSPI (but couldn't move forward due to some binary issue), Next Auth, LdapJS. I am able to authenticate my credentials using ldapjs but how to get those credentials from windows/browser is the bigger question now.

if anyone has any idea, please comment here. Thanks in advance.

**Comment by jamesperrin** [comment]
@tpanw 

I'm starting to learn Next.js, too. The majority of my environment is Windows.

Found this link using [NextAuth.js LDAP Authentication](https://next-auth.js.org/tutorials/ldap-auth-example "NextAuth.js LDAP Authentication"). Might want to give a shot.

**Comment by Shadowfita** [comment]
@tpanw , @jamesperrin  This might be a bit of a necro, but I've managed to implement NTLM auth from browser to a NextJS server (LAN) by making use of this repo [next.js-api-middleware](https://github.com/undrash/next.js-api-middleware) which allows you to create and use route middlewares similar to express webservers, and then I've made minor changes to [express-ntlm](https://github.com/einfallstoll/express-ntlm) to make it work with NextJS. 

I now have an /api/auth endpoint which issues the NTLM challenge and confirms identity against ActiveDirectory with an LDAP request and returns the users identity as a JSON response.

**Reply by Shadowfita** [reply]
I've started this project for basic NTLM auth support with a next.js middleware.ts example: https://github.com/Shadowfita/ntlm-server
